### PR TITLE
extract gateway

### PR DESCRIPTION
**Pull request by spruce-bruce**
I'm doing to clean up to split the idea of sending requests to slack off from the SlackMessage class. In order to do that safely I wanted to make sure that tests will fail when the network requests aren't sent, which they aren't currently doing.

Normally it's an anti-pattern to assert anything about the request lifecycle, but in this case there are no observable side effects of the requests to the slack api so monitoring the requests themselves is our only recourse to ensure that they're sending. That does mean that our tests are tightly coupled to the details of the slack api, but they already were because we have to mock those calls anyway.

I needed to update the `msw` package, which updated some transitive dependencies. Github actions have to have their node_modules committed so I'm going to go ahead and submit this now since that dependency change blows up the PR.

- extracting more stuff to the slack gateway
- Updates test to assert that requests are sent

### CODE REVIEW

File: src/SlackGateway.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    if (!channel) throw new Error("channel is required");
+    const response = await this._postRequest("chat.postMessage", {
+      ...message,
+      channel: channel,
+    });
+    return response;
+  }
+}
+
+module.exports = SlackGateway;
```
Comment (spruce-bruce) [inline src/SlackGateway.js:36]: This is the new slack gateway class that i'm splitting off from the main SlackMessage class.

File: src/SlackMessage.js [hunk 2]
```diff
@@ -21,37 +22,35 @@ class SlackMessageRoot {
...
   #_channel = "";
   #_ts = "";
+  #_slackGateway;
 
-  constructor(token, message = { channel: "", ts: "" }) {
+  constructor(slackToken, message = { channel: "", ts: "" }) {
     super();
-    if (!token) throw new Error("token is required");
-    this.#_token = token;
+    this.#_slackGateway = new SlackGateway(slackToken);
```
Comment (spruce-bruce) [inline src/SlackMessage.js:31]: The SlackMessage class will delegate interaction with the slack api to the SlackGateway

File: src/SlackMessage.js [hunk 3]
```diff
@@ -21,37 +22,35 @@ class SlackMessageRoot {
...
     this.#_channel = message.channel;
     this.#_ts = message.ts;
   }
 
   async _postRequest(method, body) {
     const response = await got.post(`https://slack.com/api/${method}`, {
       headers: {
-        Authorization: `Bearer ${this.#_token}`,
+        Authorization: `Bearer ${this.#_slackGateway.getToken()}`,
```
Comment (spruce-bruce) [inline src/SlackMessage.js:39]: This entire `post` call will be replaced, but for now just use the SlackGateway instance as the single source of truth for the token value.

File: test/SlackMessage.spec.js [hunk 4]
```diff
@@ -90,13 +94,21 @@ describe("SlackMessage", () => {
...
+
     await message.sendBlock({
       type: "section",
       text: {
         type: "mrkdwn",
         text: ":grey_exclamation:   api starting build",
       },
     });
+
+    await postEvents;
```
Comment (spruce-bruce) [inline test/SlackMessage.spec.js:111]: Awaiting this allows me to produce a test that fails if we call `slackMessage.sendBlock()` that doesn't result in a call to the slack api endpoint called `/api/chat.update`

**Review by spruce-bruce — COMMENTED** [review]
Annotations for reviewers

**Review by areida — APPROVED** [review]
Looks good

**Pull request merged**